### PR TITLE
codeintel: Replace encoding/json with json-iterator/go for reading LSIF

### PR DESCRIPTION
**Pull request by efritz**
Replace encoding/json with json-iterator/go for unmarshalling LSIF encoded as JSON lines. This speeds up the conversion considerably without any engineering effort.

Running the precise-code-intel integration suite's upload phase on master:

```
3m37.597s
3m35.858s
3m42.591s
```

And after this change:

```
2m59.991s
2m54.612s
2m56.651s
```

### CODE REVIEW

File: cmd/precise-code-intel-worker/internal/correlation/lsif/jsonlines/unmarshal.go [hunk 1]
```diff
@@ -4,19 +4,45 @@ import (
...
+	jsoniter "github.com/json-iterator/go"
 	"github.com/sourcegraph/sourcegraph/cmd/precise-code-intel-worker/internal/correlation/datastructures"
 	"github.com/sourcegraph/sourcegraph/cmd/precise-code-intel-worker/internal/correlation/lsif"
 )
 
+var ji = jsoniter.ConfigFastest
+
+type ID string
+
+func (id *ID) UnmarshalJSON(raw []byte) error {
```
Comment (keegancsmith) [inline cmd/precise-code-intel-worker/internal/correlation/lsif/jsonlines/unmarshal.go:19]: maybe an explanation why you need to implement UnmarshalJSON now? Does encoding/json support casting ints to strings? Or is this a new requirement/one that was previously not noticed?

Comment (efritz) [reply cmd/precise-code-intel-worker/internal/correlation/lsif/jsonlines/unmarshal.go:19]: This was just moved from another file,there's no functional change.

File: cmd/precise-code-intel-worker/internal/correlation/lsif/jsonlines/unmarshal.go [hunk 2]
```diff
@@ -4,19 +4,45 @@ import (
...
 	"fmt"
+	"strconv"
 	"strings"
 
+	jsoniter "github.com/json-iterator/go"
 	"github.com/sourcegraph/sourcegraph/cmd/precise-code-intel-worker/internal/correlation/datastructures"
 	"github.com/sourcegraph/sourcegraph/cmd/precise-code-intel-worker/internal/correlation/lsif"
 )
 
+var ji = jsoniter.ConfigFastest
```
Comment (unknwon) [inline cmd/precise-code-intel-worker/internal/correlation/lsif/jsonlines/unmarshal.go:15]: nit: can we have a better name than `ji`? :D Just feels a bit hacky.

Comment (efritz) [reply cmd/precise-code-intel-worker/internal/correlation/lsif/jsonlines/unmarshal.go:15]: Sure

**Pull request merged**